src/posix/threading.c

```diff
@@ -195,3 +223,154 @@ void sys_thread_join(sys_thread_t *thread)
     assert(thread != &_main_thread_id);
     pthread_join(thread->thread, 0);
 }
+
+sys_thread_role_t sys_thread_role(const sys_thread_t *thread)
+{
+    if (!thread)
+        thread = sys_thread_self();
+    return thread->role;
+}
+
+sys_thread_proactor_mode_t sys_thread_proactor_mode(void)
+{
+    return _self->proactor_mode;
+}
+
+sys_thread_proactor_mode_t sys_thread_proactor_set_mode(sys_thread_proactor_mode_t new_mode, void *context)
+{
+    assert(_self->role == SYS_THREAD_PROACTOR);
+    sys_thread_proactor_mode_t old_mode = _self->proactor_mode;
+    _self->proactor_mode                = new_mode;
+    _self->proactor_context             = context;
+    return old_mode;
+}
+
+void *sys_thread_proactor_context(void)
```

Comment:
In the same vein, should this be called 'sys_thread_proactor_get_context()'

Reply:
Yeah, keeping consistent with the API.